src/ray/raylet/worker_pool.cc

```diff
@@ -155,6 +149,19 @@ WorkerPool::~WorkerPool() {
   }
 }
 
+void WorkerPool::Start() {
+  if (RayConfig::instance().kill_idle_workers_interval_ms() > 0) {
+    periodical_runner_.RunFnPeriodically(
+        [this] { TryKillingIdleWorkers(); },
```

Comment:
Wouldn't this crash prestarted workers? Should we only kill idle workers that have "assigned" job ids? (or do we already do that?)

Reply:
i think this will not crash prestarted workers as long as the number of prestarted workers is less than num_workers_soft_limit_.

Reply:
The idle killer won't kill these workers, even if the total workers exceed num_workers_soft_limit_

We only kill workers whose job is dead, these workers have Nil job, and we never mark that job as dead, so the idle kill will always skip them